incipyt/project/environment.py

```diff
@@ -41,25 +75,45 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         if key in self.data:
-            raise ValueError(f"Environment variable {key} already exists, delete it before.")
-
-        logger.debug("Set environment variable %s=%s.", key, value)
+            raise ValueError(f"Environment variable {key} is set already, delete it before.")
+        logger.debug("set %s=%s.", key, value)
         self.data[key] = value
 
     def _prompt(self, key):
-        if key in variables.metadata and variables.metadata[key].do_not_prompt:
-            user_input = variables.metadata[key].default
-        else:
-            user_input = click.prompt(
-                key.replace("_", " ").lower().capitalize(),
-                default=variables.metadata[key].default if key in variables.metadata else "",
-                type=str,
-            )
-        if user_input:
-            return user_input
-        else:
-            return (
-                user_input
-                if key in variables.metadata and variables.metadata[key].required
-                else None
-            )
+        if not (var := variables.get(key)):
```

Comment:
Using warlus means you only intent to use the variable inside of the block, you use it outside, thus is should be defined outiside